roles/kubernetes/preinstall/tasks/main.yml

```diff
@@ -60,11 +60,11 @@
 
 - name: check cloud_provider value
   fail:
-    msg: "If set the 'cloud_provider' var must be set either to 'generic', 'gce', 'aws' or 'openstack'"
-  when: cloud_provider is defined and cloud_provider not in ['generic', 'gce', 'aws', 'openstack']
+    msg: "If set the 'cloud_provider' var must be set either to 'generic', 'gce', 'aws', 'openstack' or 'vsphere'"
+  when: cloud_provider is defined and cloud_provider not in ['generic', 'gce', 'aws', 'openstack', 'vsphere']
 
-- include: openstack-credential-check.yml
-  when: cloud_provider is defined and cloud_provider == 'openstack'
+- include: "{{ cloud_provider}}-credential-check.yml"
+  when: cloud_provider is defined and (cloud_provider == 'openstack' or cloud_provider == 'vsphere')
```

Comment:
nit: you can also do cloud_provider in ['openstack', 'vsphere']. It's a little more compact